### PR TITLE
s3: Allow authentication using session token

### PR DESCRIPTION
**Pull request by isidentical**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by isidentical** [comment]
Docs: iterative/dvc.org#2084

File: dvc/tree/s3.py [hunk 1]
```diff
@@ -68,16 +69,23 @@ def s3(self):
...
         }
+        config = None
 
         if self.access_key_id:
             session_opts["aws_access_key_id"] = self.access_key_id
         if self.secret_access_key:
             session_opts["aws_secret_access_key"] = self.secret_access_key
+        if self.session_token:
+            session_opts["aws_session_token"] = self.session_token
+            config = boto3.session.Config(signature_version="s3v4")
```
Comment (shcheklein) [inline dvc/tree/s3.py:80]: out of curiosity - is it required? is it always `s3v4`?

Comment (isidentical) [reply dvc/tree/s3.py:80]: I've just checked the docs, yes it is. Thanks for catching it!

**Review by efiop — COMMENTED** [review]
Looks great! Let's add a simple unittest to tests/unit/tree/test_s3.py

The next convenience feature would be https://github.com/iterative/dvc/issues/5241 :slightly_smiling_face:

**Review by efiop — APPROVED** [review]

**Pull request merged**